salt/_modules/docker_registry.py

```diff
@@ -0,0 +1,85 @@
+'''
+Functions to interact with a Docker registry.
+'''
+
+import logging
+
+import salt.utils.json
+import salt.utils.path
+
+log = logging.getLogger(__name__)
+
+
+__virtualname__ = 'docker_registry'
+
+
+def __virtual__():
+    return __virtualname__
+
+
+def get_image_manifest(image, tls_verify=True):
+    '''
+    Retrieve an image manifest from a repository.
+
+    .. note::
+
+       This uses the :command:`skopeo` command.
+
+    image
+        Name of the image (e.g. `localhost:5000/metalk8s/nginx:1.14.2`)
```

Comment:
minor: it's more a URL or URI than a name

Reply:
I believe `name` is the nomenclature used in Docker spheres. It's definitely not a URL.

Reply:
i thought we should tag the image with the registry URL to be able to push it no ?

Reply:
The first part (e.g. `docker.io/` or `localhost:5000/`) is not a URL, the OCI Distribution spec is much more hairy than that. The URL of the repository entrypoint would be `https://docker.io/v2` or `http://localhost:5000/v2`.